build.gradle

```diff
@@ -431,39 +402,95 @@ configure(javaProjects) {
         }
     }
 
-    // Compile the Thrift service definitions.
-    task compileThrift(group: 'Build',
-                       description: 'Compiles the .thrift files') {
-        [ 'main', 'test' ].each { scope ->
-            def inputDir = "${project.projectDir}/src/${scope}/thrift"
-            def outputDir = "${project.projectDir}/src/${scope}/gen-java"
-
-            if (project.file(inputDir).isDirectory()) {
-                inputs.dir inputDir
-                outputs.dir outputDir
-
-                project.sourceSets[scope].java.srcDir outputDir
-
-                doLast {
-                    project.fileTree(inputDir) {
-                        include '**/*.thrift'
-                    }.each { sourceFile ->
-                        project.mkdir(outputDir)
-                        project.exec {
-                            commandLine project.ext.thriftPath,
-                                    '-gen', 'java',
-                                    '-out', outputDir,
-                                    '-I', "${sourceFile.parentFile.absolutePath}",
-                                    sourceFile.absolutePath
+    // TODO(anuraaga): Consider replacing with https://github.com/yodle/griddle
+    afterEvaluate {
```

Comment:
Could you explain why this had to be moved into `afterEvaluate { .. }`?

Reply:
Gives a chance for other plugins (ike JMH) to add source sets, which we use here. I guess in general, top-level build logic often needs to go in afterevaluate to let the individual projects do their thing first.